core/src/main/java/com/sap/ai/sdk/core/AiCoreServiceStub.java

```diff
@@ -0,0 +1,98 @@
+package com.sap.ai.sdk.core;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.PropertyAccessor;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.collect.Iterables;
+import com.sap.cloud.sdk.cloudplatform.connectivity.ApacheHttpClient5Accessor;
+import com.sap.cloud.sdk.cloudplatform.connectivity.Destination;
+import com.sap.cloud.sdk.services.openapi.apiclient.ApiClient;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import org.springframework.http.client.BufferingClientHttpRequestFactory;
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+
+/** Container for an API client and destination. */
+@FunctionalInterface
+public interface AiCoreServiceStub {
```

Comment:
The naming `Stub` sounds a little bit like test code. I get that we don't want to unnecessarily make customer code more verbose by having the class be named `DefaultAiCoreService`. We could consider `AiCoreServiceI` or maybe simply `AiService`. No strong opinion.

Reply:
Current suggestion is:
```
AiCoreService extends AiCoreDestination
AiCoreDeployment extends AiCoreDestination
```

> Why not `OrchestrationClient` as suggested [below](https://github.com/SAP/ai-sdk-java/pull/79#pullrequestreview-2340551315)?

Because it would conflict with `OpenAI` client (and future) use cases-

> Why "Destination"?

Because that's the primary value to extract, via `destination()`. Allowing `client()` is only for convenience. Furthermore we may challenge the whole _SAP Cloud SDK_ type `ApiClient` related to that, in mid future - before release maybe.